src/scanpy/preprocessing/_simple.py

```diff
@@ -720,14 +791,29 @@ def regress_out(
         regres = regressors_chunk[idx] if variable_is_categorical else regressors
         tasks.append(tuple((data_chunk, regres, variable_is_categorical)))
 
-    from joblib import Parallel, delayed
+    res = None
+    if not variable_is_categorical:
```

Comment:
I think that the implementation as both working. Because we clear the tests that are out there. Because in the end we test `variable_is_categorical` and the algorithm is the same.